MediaWikiKit/MediaWikiKit/MWKArticle.m

```diff
@@ -570,9 +570,21 @@ - (NSAttributedString*)summaryHTML {
         if (!nonEmptyParagraphsWithSelectedChildren.count) {
             continue;
         }
-        NSString* lineSeparatedParagraphs = [nonEmptyParagraphsWithSelectedChildren componentsJoinedByString:@"<br/><br/>"];
-        NSData* xpathData                 = [lineSeparatedParagraphs dataUsingEncoding:NSUTF8StringEncoding];
-        return [[NSAttributedString alloc] initWithHTMLData:xpathData site:self.site];
+        NSString* lineSeparatedParagraphs = [nonEmptyParagraphsWithSelectedChildren componentsJoinedByString:@"</p><p>"];
```

Comment:
this is and the next line could be replace by a map which wraps in paragraph in `<p></p>`. makes the intent clearer, IMHO. also, I'd do the wrapping with array joining instead of string formatting (which I'm guessing would be a hair faster): `[@["<p>", paragraphChildren, "</p>"] componentsSeparatedByString:@""]`.

Also, this would be much simpler written in Swift.  Just sayin :wink: 

``` swift
nonEmptyParagraphsWithSelectedChildren.map() { "<p>\($0)</p>" }
```